x-pack/plugins/lens/public/visualizations/metric/dimension_editor.tsx

```diff
@@ -131,7 +131,7 @@ function MaximumEditor({ setState, state, idPrefix }: SubProps) {
 }
 
 function SecondaryMetricEditor({ accessor, idPrefix, frame, layerId, setState, state }: SubProps) {
-  const columnName = getColumnByAccessor(accessor, frame.activeData?.[layerId].columns)?.name;
+  const columnName = getColumnByAccessor(accessor, frame.activeData?.[layerId]?.columns)?.name;
```

Comment:
Nice! This is why we enabled `noUncheckedIndexedAccess ` in `@elastic/charts` https://github.com/elastic/elastic-charts/pull/2006.

That way any array access is treated as a union with `undefined` unless the type is an explicit tuple.

Reply:
Good idea! It'd be good to add it in Kibana, we had some bugs in the past caused by it.